test/runtests.jl

```diff
@@ -118,6 +118,16 @@ ch2 = chain(1:0, 1:2:5, 0.2:0.1:1.6)
 @test eltype(ch2) == typejoin(Int, Float64)
 @test collect(ch2) == [1:2:5; 0.2:0.1:1.6]
 
+ch3 = chain(1:10,1:10, 1:10)
+@test length(ch3) == 30
+@test Base.iteratorsize(ch3) == Base.HasLength()
```

Comment:
looks like I introduced some typo at the last moment just before pushing it (Meth instead of MethodErrod), that was causing the failure

Base.iteratorsize is doing the right job on my machine, i.e. using the method defined in Iterators.jl. Should I still change the test?